src/Register.hs

```diff
@@ -17,16 +17,22 @@ not :: Register -> Register
 not = fmap Bit.not
 
 and :: Register -> Register -> Register
-and x y = fromList (fmap (uncurry Bit.and) (zip (toList x) (toList y)))
-
--- or :: Register -> Register -> Register
---
--- xor :: Register -> Register -> Register
---
--- nand :: Register -> Register -> Register
---
--- nor :: Register -> Register -> Register
---
--- xnor :: Register -> Register -> Register
---
--- implies :: Register -> Register -> Register
+and x y = fromList (uncurry Bit.and `fmap` zip (toList x) (toList y))
+
+or :: Register -> Register -> Register
+or x y = fromList (uncurry Bit.or `fmap` zip (toList x) (toList y))
+
+xor :: Register -> Register -> Register
+xor x y = fromList (uncurry Bit.xor `fmap` zip (toList x) (toList y))
+
+nand :: Register -> Register -> Register
+nand x y = fromList (uncurry Bit.nand `fmap` zip (toList x) (toList y))
+
+nor :: Register -> Register -> Register
+nor x y = fromList (uncurry Bit.nor `fmap` zip (toList x) (toList y))
+
+xnor :: Register -> Register -> Register
+xnor x y = fromList (uncurry Bit.xnor `fmap` zip (toList x) (toList y))
+
+implies :: Register -> Register -> Register
+implies x y = fromList (uncurry Bit.implies `fmap` zip (toList x) (toList y))
```

Comment:
The operator form of `` `fmap` `` is `<$>`
Also consider using `fromList $` to remove one level of parentheses.